### PR TITLE
added size failure check

### PR DESCRIPTION
**Pull request by jadhavrohit924**
#### Problem
What is being fixed?  Examples:
* Addressed https://github.com/project-chip/connectedhomeip/issues/14304 and handle if buffer is small

#### Change overview
Handle if buffer is small

#### Testing
manual test

### CODE REVIEW

**Comment by jadhavrohit924** [comment]
@bzbarsky-apple please review

**Review by msandstedt — APPROVED** [review]

File: src/app/clusters/ota-requestor/OTARequestor.cpp [hunk 1]
```diff
@@ -584,8 +590,6 @@ CHIP_ERROR OTARequestor::ExtractUpdateDescription(const QueryImageResponseDecoda
     update.softwareVersion = response.softwareVersion.Value();
 
     VerifyOrReturnError(response.updateToken.HasValue(), CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrReturnError(response.updateToken.Value().size() >= 8 && response.updateToken.Value().size() <= 32,
```
Comment (carol-apple) [inline src/app/clusters/ota-requestor/OTARequestor.cpp:587]: I think we should still check for the size, no? This guarantees that by the time we do the `CopySpanToMutableSpan` above, we are definitely not going to copy more than the buffer can handle.

File: src/app/clusters/ota-requestor/OTARequestor.cpp [hunk 2]
```diff
@@ -128,7 +128,13 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
             ChipLogDetail(SoftwareUpdate, "Update available from %" PRIu32 " to %" PRIu32 " version",
                           requestorCore->mCurrentVersion, update.softwareVersion);
             MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
-            CopySpanToMutableSpan(update.updateToken, updateToken);
+            // This function copies the bytespan to mutablebytespan only if size of mutablebytespan buffer is greater or equal to
+            // bytespan otherwise we are copying data upto available size.
+            err = CopySpanToMutableSpan(update.updateToken, updateToken);
+            if (err == CHIP_ERROR_BUFFER_TOO_SMALL)
+            {
+                memcpy(updateToken.data(), update.updateToken.data(), updateToken.size());
```
Comment (bzbarsky-apple) [inline src/app/clusters/ota-requestor/OTARequestor.cpp:136]: So why is it ok to drop the part of the update token that does not fit?

File: src/app/clusters/ota-requestor/OTARequestor.cpp (hunk 1 above)
Comment (jadhavrohit924) [reply src/app/clusters/ota-requestor/OTARequestor.cpp:587]: In CopySpanToMutableSpan we are checking size of buffer

File: src/app/clusters/ota-requestor/OTARequestor.cpp (hunk 2 above)
Comment (jadhavrohit924) [reply src/app/clusters/ota-requestor/OTARequestor.cpp]: I emptied buffer and gererated updateToken. In specs it is mentioned that use nodeId if we don’t  have token.

Comment (carol-apple) [reply src/app/clusters/ota-requestor/OTARequestor.cpp]: I don't think that's the right approach. We should only use the nodeId as token if we've lost the token. In this case, the token is invalid and it feels like we should treat it as such. We should error out here or in the original location (where you removed in this PR).

@tcarmelveilleux thoughts?

Comment (jadhavrohit924) [reply src/app/clusters/ota-requestor/OTARequestor.cpp]: @carol-apple This is a outdated code. Pl check here https://github.com/project-chip/connectedhomeip/commit/50a2383577dafe62a958184f4deec2bdc744cde9#:~:text=requestorCore%2D%3EGenerateUpdateToken()%3B

**Review by andy31415 — APPROVED** [review]

**Pull request merged**